Firebase/Messaging/FIRMessaging.m

```diff
@@ -611,7 +611,8 @@ - (BOOL)shouldBeConnectedAutomatically {
   // We require a token from Instance ID
   NSString *token = self.defaultFcmToken;
   // Only on foreground connections
-  UIApplicationState applicationState = [UIApplication sharedApplication].applicationState;
+  UIApplication *application = FIRMessagingUIApplication();
+  UIApplicationState applicationState = application.applicationState;
```

Comment:
This may return `UIApplicationStateActive` for extensions regardless of the host application state. Is that the correct behavior?

Reply:
We haven't finalized the official support for extensions so any function is called in extensions is not guarantee to be working. This CL just to ensure compilation works.

Reply:
If developer wants to establish a MCS connection in app extensions, it should probably work so the application state is for extension app.

Reply:
Actually no, the application object would be nil, so there's no way to check the application state in extensions. Basically we should return no for this case.

I'm also adding nil check for all the cases here.